docs/quickstart/kubernetes.md

```diff
@@ -0,0 +1,167 @@
+```
+
+Port forward Zeppelin server port,
+ 
+```
+kubectl port-forward zeppelin-server 8080:80
+```
+
+and browse [localhost:8080](http://localhost:8080).
+
+
+To shutdown,
+
+```
+kubectl delete -f zeppelin-server.yaml
+```
+
+
+## Spark Interpreter
+
+Build spark docker image to use Spark Interpreter.
+Download spark binary distribution and run following command.
+Spark 2.4.0 or later version is required.
+
+```
+# if you're using minikube, set docker-env
+$ eval $(minikube docker-env)
+
+# build docker image
+$ <spark-distribution>/bin/docker-image-tool.sh -m -t 2.4.0 build
+```
+
+Run `docker images` and check if `spark:2.4.0` is created.
+Configure `sparkContainerImage` of `zeppelin-server-conf` ConfigMap in `zeppelin-server.yaml`.
+
+
+Create note and configure executor number (default 1)
+
+```
+%spark.conf
+spark.executor.instances  5
+``` 
+
+And then start your spark interpreter
+
+```
+%spark
+sc.parallelize(1 to 100).count
+...
+```
+While `master` property of SparkInterpreter starts with `k8s://` (default `k8s://https://kubernetes.default.svc` when Zeppelin started using zeppelin-server.yaml), Spark executors will be automatically created in your Kubernetes cluster.
+Spark UI is accessible by clicking `SPARK JOB` on the Paragraph. 
+
+Check [here](https://spark.apache.org/docs/latest/running-on-kubernetes.html) to know more about Running Spark on Kubernetes.
+
+
+## Build Zeppelin image manually
+
+To build your own Zeppelin image, first build Zeppelin project with `-Pbuild-distr` flag.
+
+```
+$ mvn package -DskipTests -Pbuild-distr <your flags>
+```
+
+Binary package will be created under `zeppelin-distribution/target` directory. Move created package file under `scripts/docker/zeppelin/bin/` directory.
+
+```
+$ mv zeppelin-distribution/target/zeppelin-*.tar.gz scripts/docker/zeppelin/bin/
+```
+
+`scripts/docker/zeppelin/bin/Dockerfile` downloads package from internet. Modify the file to add package from filesystem.
+
+```
+...
+
+# Find following section and comment out
+#RUN echo "$LOG_TAG Download Zeppelin binary" && \
+#    wget -O /tmp/zeppelin-${Z_VERSION}-bin-all.tgz http://archive.apache.org/dist/zeppelin/zeppelin-${Z_VERSION}/zeppelin-${Z_VERSION}-bin-all.tgz && \
+#    tar -zxvf /tmp/zeppelin-${Z_VERSION}-bin-all.tgz && \
+#    rm -rf /tmp/zeppelin-${Z_VERSION}-bin-all.tgz && \
+#    mv /zeppelin-${Z_VERSION}-bin-all ${Z_HOME}
+
+# Add following lines right after the commented line above
+ADD zeppelin-${Z_VERSION}.tar.gz /
+RUN ln -s /zeppelin-${Z_VERSION} /zeppelin
+...
+```
+
+Then build docker image.
+
+```
+# configure docker env, if you're using minikube
+$ eval $(minikube docker-env) 
+
+# change directory
+$ cd scripts/docker/zeppelin/bin/
+
+# build image. Replace <tag>.
+$ docker build -t <tag> .
+```
+
+Finally, set custom image `<tag>` just created to `image` and `ZEPPELIN_K8S_CONTAINER_IMAGE` env variable of `zeppelin-server` container spec in `zeppelin-server.yaml` file.
```

Comment:
Probably one small section that lists the current limitations will be useful. The one which is probable worth highlighting is about the container image used by the interpreter other than spark. It will be the same as Zeppelin container image.

Reply:
Added more information below.

Reply:
This information is also very helpful